dhis-2/dhis-api/src/main/java/org/hisp/dhis/event/EventStatus.java

```diff
@@ -52,17 +53,9 @@ public int getValue() {
     return value;
   }
 
-  public static EventStatus fromInt(int status) {
-    for (EventStatus eventStatus : EventStatus.values()) {
-      if (eventStatus.getValue() == status) {
-        return eventStatus;
-      }
-    }
+  public static final Set<EventStatus> ALLOW_DATA_VALUES_STATUSES =
+      Set.of(ACTIVE, VISITED, COMPLETED);
 
-    throw new IllegalArgumentException();
-  }
-
-  public static boolean isExistingEvent(EventStatus status) {
-    return status != null && (COMPLETED.equals(status) || VISITED.equals(status));
-  }
+  public static final Set<EventStatus> NO_ALLOW_DATA_VALUES_STATUSES =
```

Comment:
I'd call this one `DISALLOWED_DATA_VALUES_STATUSES`, and in case the comment above makes sense, just `DISALLOWED_STATUSES`

Reply:
I think not allowed and disallowed are not synonyms, but we could ask a native speaker about it.
The idea here is the same as above, these statuses are valid, the just do not permit to define data values.
Maybe because this was extracted from a bigger PR is not clear, but we will add another validation that is simply going to fail if the event has a status in `NO_ALLOW_DATA_VALUES_STATUSES` and `dataValues` is not empty

Reply:
`STATUSES_DISALLOWING_DATA_VALUES` or `STATUSES_WITHOUT_DATA_VALUES`